### PR TITLE
[ContainerApp] Update Cloud Build code to the ARM simplification

### PR DESCRIPTION
**Pull request by daniv-msft**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### Related command
az containerapp up
az containerapp create
az containerapp update

### General Guidelines

- [x] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [x] Have you run `python scripts/ci/test_index.py -q` locally? (`pip install wheel==0.30.0` required)
- [x] My extension version conforms to the [Extension version schema](https://github.com/Azure/azure-cli/blob/release/doc/extensions/versioning_guidelines.md)

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your pull request is merged into main branch, a new pull request will be created to update `src/index.json` automatically.  
You only need to update the version information in file setup.py and historical information in file HISTORY.rst in your PR but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
ContainerApp

**Comment by daniv-msft** [comment]
Tagging @harryli0108 for FYI

**Comment by daniv-msft** [comment]
@harryli0108 Could you please take care of updating/merging this PR from there as I won't be there to do so?

**Review by Moazzem-Hossain-pixel — COMMENTED** [review]
I would like to ask support for resolving my issues if possible!

**Comment by yanzhudd** [comment]
Please note that Azure CLI team is only responsible for code style and specification, but not business logic. It would be better to involve the service team to review this PR first.

**Comment by yanzhudd** [comment]
Please add test for this change to check if it can meet your expectation.

**Comment by Moazzem-Hossain-pixel** [comment]
How may get support in these issues? If there is an opportunity to help me that will be highly appreciated.

Kind Regards

File: src/containerapp/HISTORY.rst [hunk 1]
```diff
@@ -11,6 +11,7 @@ upcoming
 * 'az containerapp env telemetry otlp remove': Support remove environment otlp configuration with --otlp-name
 * 'az containerapp env telemetry otlp show': Support show environment otlp configuration with --otlp-name
 * 'az containerapp env telemetry otlp list': Support show environment otlp configurations
+* 'az containerapp up/create/update': Update Cloud Build code to use the simplified ARM calling pattern
```
Comment (Greedygre) [inline src/containerapp/HISTORY.rst:14]: Please move this to `upcoming`

**Comment by Greedygre** [comment]
Hi @Moazzem-Hossain-pixel 

Please get approval from your team who could be responsible for your business logic firstly.

File: src/containerapp/azext_containerapp/_cloud_build_utils.py [hunk 2]
```diff
@@ -134,14 +129,17 @@ def spin():
...
             archive_source_code(data_file_path, source)
             done_spinner = True
             thread.join()
 
         # File upload
         done_spinner = False
         thread = display_spinner("Uploading data")
+        container_app_id = container_app_result["id"]
+        base_proxy_endpoint = container_app_result["properties"]["eventStreamEndpoint"].rstrip("/eventstream")
+        upload_endpoint = base_proxy_endpoint + f"/upload?token={token}"
```
Comment (Greedygre) [inline src/containerapp/azext_containerapp/_cloud_build_utils.py:142]: Here can be:
upload_endpoint =f"{base_proxy_endpoint}/upload?token={token}"

File: src/containerapp/azext_containerapp/_cloud_build_utils.py [hunk 3]
```diff
@@ -159,6 +157,9 @@ def spin():
                 os.unlink(data_file_path)
         if not response_file_upload.ok:
             raise ValidationError(f"Error when uploading the file, request exited with {response_file_upload.status_code}")
+        file_upload_json = response_file_upload.json()
+        build_name = file_upload_json["name"]
+        log_streaming_endpoint = base_proxy_endpoint + f"/builds/{build_name}/logstream"
```
Comment (Greedygre) [inline src/containerapp/azext_containerapp/_cloud_build_utils.py:162]: Same as above.

**Comment by Greedygre** [comment]
Hi @Moazzem-Hossain-pixel 
Do you have any tests to cover these changes?
Why are these changes needed?

Please rerun related tests to make sure e2e tests can pass. You can update the related recording files or paste your test rerun result in the description.

**Comment by Greedygre** [comment]
> Hi @Moazzem-Hossain-pixel Do you have any tests to cover these changes? Why are these changes needed?
> 
> Please rerun related tests to make sure e2e tests can pass. You can update the related recording files or paste your test rerun result in the description.

Run related test with command: `azdev test {your-command-file/your-command-test-case} --live`
About `azdev test`, refer to: https://github.com/Azure/azure-cli/blob/dev/doc/configuring_your_machine.md

**Comment by harryli0108** [comment]
Continuing to contribute to this change in a new PR #7522. Please feel free to close this PR if needed

**Pull request closed without merging**